.github/workflows/lint.yml

```
@@ -0,0 +1,302 @@
+        with:
+          toolchain: stable
+          cache-on-failure: true
+      - name: Ensure no arbitrary or proptest dependency on default build
+        run: cargo tree --package zebrad -e=features,no-dev | grep -Eq "arbitrary|proptest" && exit 1 || exit 0
+
+  # Checks that selected rates can compile with power set of features
+  features:
+    name: features
+    permissions:
+      id-token: write
+      statuses: write
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 #v5.0.0
+        with:
+          persist-credentials: false
+
+      - uses: actions-rust-lang/setup-rust-toolchain@ab6845274e2ff01cd4462007e1a9d9df1ab49f42 #v1.14.0
+        with:
+          toolchain: nightly
+          cache-on-failure: true
+      - uses: arduino/setup-protoc@c65c819552d16ad3c9b72d9dfd5ba5237b9c906b #v3.0.0
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: cargo install cargo-hack
+        uses: taiki-e/install-action@3ee5d63d29478156148c0b53e9f3447829b47bc2 #v2.58.23
+        with:
+          tool: cargo-hack
+      - run: cargo hack check --all
+        env:
+          RUSTFLAGS: -D warnings
+
+  check-cargo-lock:
+    name: check-cargo-lock
+    permissions:
+      id-token: write
+      statuses: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 #v5.0.0
+        with:
+          persist-credentials: false
+      - uses: actions-rust-lang/setup-rust-toolchain@ab6845274e2ff01cd4462007e1a9d9df1ab49f42 #v1.14.0
+        with:
+          toolchain: stable
+          cache-on-failure: true
+      - uses: arduino/setup-protoc@c65c819552d16ad3c9b72d9dfd5ba5237b9c906b #v3.0.0
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - run: cargo check --locked --all-features --all-targets
```

Again, no need for `cargo check`

---

Here the difference is that with Clippy we never pass the `--locked` flag (_should we?_), and the separate `cargo check` runs once with `--locked --all-features --all-targets` which:
- Fails fast if `Cargo.lock` is out of sync or would need to be regenerated, and 
- Guarantees the crate builds when every feature is enabled simultaneously (this is skipped in the Clippy runs).

---

Ah yes, that makes sense. We should use `--locked` when building zebra but not use it when building the crates individually, so it's fine as is